fastlane_bot/events/async_event_update_utils.py

```diff
@@ -317,12 +306,16 @@ def process_contract_chunks(
         func: Callable,
         df_combined: pd.DataFrame = None,
         read_only: bool = False,
+        carbon_v1_forks: List[str] = None,
 ) -> pd.DataFrame:
     lst = []
     # write chunks to csv
     for idx, chunk in enumerate(chunks):
         loop = asyncio.get_event_loop()
-        df = loop.run_until_complete(func(chunk))
+        if filename == "tokens_and_fee_df.csv":
+            df = loop.run_until_complete(func(chunk, carbon_v1_forks))
+        else:
+            df = loop.run_until_complete(func(chunk))
```

Comment:
I see no advantage in passing `func` as input to this function, so I recommend just calling the related functions directly (`main_get_tokens_and_fee` in the 'if' and `main_get_missing_tkn` in the 'else').

If these two functions had an identical interface (input and output), then there would be some advantage in doing so (but even under such scenario, the filename-based decision on which function to call, would still reduce that advantage to a minimum).

Side note: since both of these functions are used only in this module, you can also prepend the name of each one of them with an underscore.

Reply:
The original interface was the same, and thanks to my most recent changes they are now the same again, so this has been corrected